tests/internal/test_tracer_flare.py

```diff
@@ -93,89 +101,139 @@ def test_single_process_partial_failure(self):
         assert os.path.exists(self.flare_file_path)
         assert not os.path.exists(self.config_file_path)
 
-        self.flare.send(self.mock_flare_send_request)
+        self.flare.send(MOCK_FLARE_SEND_REQUEST)
+
+    def test_no_app_logs(self):
+        """
+        Validate that app logs are not being added to the
+        file, just the tracer logs
+        """
+        app_logger = Logger(name="my-app", level=DEBUG_LEVEL_INT)
+        self.flare.prepare("DEBUG")
+
+        app_log_line = "this is an app log"
+        app_logger.debug(app_log_line)
+
+        assert os.path.exists(self.flare_file_path)
+
+        with open(self.flare_file_path, "r") as file:
+            for line in file:
+                assert app_log_line not in line, f"File {self.flare_file_path} contains excluded line: {app_log_line}"
+
+        self.flare.clean_up_files()
+        self.flare.revert_configs()
+
+    def confirm_cleanup(self):
+        assert not self.flare.flare_dir.exists(), f"The directory {self.flare.flare_dir} still exists"
+        assert self._get_handler() is None, "File handler was not removed"
+
+
+class TracerFlareMultiprocessTests(unittest.TestCase):
+    def setUp(self):
+        if sys.version_info >= (3, 10):
+            self.shared_dir = tempfile.TemporaryDirectory(ignore_cleanup_errors=True)
+        else:
+            # There will be a clean up error for python < 3.10 that can be disregarded
+            self.shared_dir = tempfile.TemporaryDirectory()
+        self.errors = multiprocessing.Queue()
+
+    def tearDown(self):
+        try:
+            self.shared_dir.cleanup()
+        except Exception:
+            # This will always fail because our Flare logic cleans up the entire directory
+            # We're explicitly calling this in tearDown so that we can remove
+            # the error log clutter for python < 3.10
+            pass
 
     def test_multiple_process_success(self):
         """
         Validate that the tracer flare will generate for multiple processes
         """
         processes = []
         num_processes = 3
-
-        def handle_agent_config():
-            self.flare.prepare("DEBUG")
-
-        def handle_agent_task():
-            self.flare.send(self.mock_flare_send_request)
+        flares = []
+        for _ in range(num_processes):
+            flares.append(
+                Flare(
+                    trace_agent_url=TRACE_AGENT_URL,
+                    flare_dir=pathlib.Path(self.shared_dir.name),
+                    ddconfig={"config": "testconfig"},
+                )
+            )
```

Comment:
I realized if we have multiple processes because of multiple tracers, they would also each have their own `Flare`, so I added this